firmware/application/apps/ui_ss_viewer.cpp

```diff
@@ -96,4 +96,41 @@ void ScreenshotViewer::paint(Painter& painter) {
     }
 }
 
-}  // namespace ui
\ No newline at end of file
+SplashViewer::SplashViewer(
+    NavigationView& nav,
+    const std::filesystem::path& path)
+    : nav_{nav},
+      path_{path} {
+    valid_image = false;
+    set_focusable(true);
+}
+
+bool SplashViewer::on_key(const KeyEvent key) {
+    if (valid_image && key == KeyEvent::Right) {
+        delete_file(splash_dot_bmp);
+        copy_file(path_, splash_dot_bmp);
+    }
+
+    nav_.pop();
+    return true;
+}
+
+void SplashViewer::paint(Painter& painter) {
+    constexpr size_t pixel_width = 240;
```

Comment:
I added `screen_width` and `screen_height` constants in ui_navigation.hpp (I think?), fine for now, but I do want to go and replace all these with those values.

Reply:
Updated file